android/greengrass-nucleus-android/src/main/java/com/aws/greengrass/android/managers/AndroidBaseApkManager.java

```diff
@@ -0,0 +1,698 @@
+        }
+        PendingIntent pendingIntent = PendingIntent.getBroadcast(context, 0, intent, flag);
+        PackageInstaller packageInstaller = context.getPackageManager().getPackageInstaller();
+        IntentSender statusReceiver = pendingIntent.getIntentSender();
+
+        OperationContext uninstallContext = new OperationContext(logger);
+        pendingRequests.put(requestId, uninstallContext);
+        try {
+            synchronized (uninstallContext) {
+                packageInstaller.uninstall(packageName, statusReceiver);
+
+                uninstallContext.wait();
+
+                Integer status = uninstallContext.status;
+                if (status == null) {
+                    throw new IOException("Uninstall failed, status unknown");
+                }
+
+                if (status != PackageInstaller.STATUS_SUCCESS) {
+                    if (uninstallContext.message != null) {
+                        throw new IOException("Uninstall failed, status " + status + " message "
+                                + uninstallContext.message);
+                    } else {
+                        throw new IOException("Uninstall failed, status " + status);
+                    }
+                }
+            }
+        } finally {
+            pendingRequests.remove(requestId);
+        }
+    }
+
+    private IntentFilter getIntentFilter() {
+        IntentFilter intentFilter = new IntentFilter();
+        intentFilter.addAction(PACKAGE_INSTALL_STATUS_ACTION);
+        intentFilter.addAction(PACKAGE_UNINSTALL_STATUS_ACTION);
+        return intentFilter;
+    }
+
+    /**
+     * Handle result of install or uninstall.
+     *
+     * @param intent information about install or uninstall operation status.
+     * @param operation "install" or "uninstall" operation is reported
+     */
+    private void handleOperationResult(Intent intent, @NonNull String operation) {
+        Bundle extras = intent.getExtras();
+        int status = extras.getInt(PackageInstaller.EXTRA_STATUS);
+        String message = extras.getString(PackageInstaller.EXTRA_STATUS_MESSAGE);
+        String packageName = extras.getString(EXTRA_PACKAGE_NAME);
+        String requestId = extras.getString(EXTRA_REQUEST_ID);
+        Logger logger = getLogger(requestId);
+        // TODO: set status specific for package name instead of generic
+        switch (status) {
+            case PackageInstaller.STATUS_PENDING_USER_ACTION:
+                // This app isn't privileged, so the user has to confirm the uninstall.
+                Intent confirmIntent = (Intent) extras.get(Intent.EXTRA_INTENT);
+                confirmIntent.addFlags(FLAG_ACTIVITY_NEW_TASK);
+                logger.atDebug().log("Requesting confirmation from user to {} package", operation);
+                Context context = contextProvider.getContext();
+                if (isIntentResolvable(confirmIntent, context)) {
+                    context.startActivity(confirmIntent);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

When launching an Android activity with `startActivity(Intent, ...)`, confirm that an application on the device can receive the implicit intent and launch the activity. If an application cannot, then your application crashes when `startActivity` is called. See the following diffs on GitHub where this issue was fixed: [diff1](https://github.com/jtribe/shelly/commit/a51e6db04fe01da362d168465979461d0893529f#diff-6a81c9af078ea62d2a817e1b8de61a4f190d90530923d26d712eebf6c14dba62L113), [diff2](https://github.com/didi/DoraemonKit/commit/b6815f63159a564d3110242d78611fb61bdbf9b4#diff-86123abc1b9085ef0d59807e11f3aa9c0a0f0d138f7543dce034014c5276aa9fL89)

Learn more: [Common Intents in the Android developer guide](https://developer.android.com/guide/components/intents-common)